src/modules/languages/rust.nix

```diff
@@ -13,67 +10,91 @@ let
           nixpkgs:
             follows: nixpkgs
   '';
+
+  error = dbg: "To use languages.rust.${dbg}, you need to add the following to your devenv.yaml:\n\n${setup}";
 in
 {
   options.languages.rust = {
     enable = lib.mkEnableOption "tools for Rust development";
 
-    packages = lib.mkOption {
+    components = lib.mkOption {
+      type = lib.types.listOf lib.types.str;
+      default = [ "rustc" "cargo" "clippy" "rustfmt" "rust-analyzer" ];
+      defaultText = lib.literalExpression ''[ "rustc" "cargo" "clippy" "rustfmt" "rust-analyzer" ]'';
+      description = ''
+        List of [Rustup components](https://rust-lang.github.io/rustup/concepts/components.html)
+        to install. Defaults to those available in `nixpkgs`.
+      '';
+    };
```

Comment:
Hm, what's supposed to happen here if I set `components = [ "rustc" "cargo" ]`? Is this supposed to work like components or additional components? Right now it works like additional components. There's no way to omit e.g. "rust-analyzer". Not saying we need to support this, but at the very least it should be documented as such.

Reply:
Disregard this, made a config type when testing.